### DISCUSSION TITLE
what's the best way to configure a hid device?

### DISCUSSION THREAD

**Discussion by dbbbit** [hathach/tinyusb#1424] [General]
I have a joystick device and i would like to configure it by custom windows application. 
 I realized it by using two interface,  one for joystick, and another for in/out custom configure
It works fine, but some game would identify it as two device with same name, which confuse user.
So i try to fine some new way to avoid this problem, is  there any example for HID set/get feature? 
Or any other way to configure a joystick with in/out pipe?

**Comment by hathach** [comment]
you could use the set/get output report via control transfer with different report ID for your configuration. There is callback in hid composite example
